packages/dev/core/src/Materials/Textures/internalTexture.ts

```diff
@@ -277,6 +277,12 @@ export class InternalTexture extends TextureSampler {
         this._uniqueId = id;
     }
 
+    /**
+     * Gets the Engine the texture belongs to.
+     * @returns The babylon engine
+     */
+    public getEngine<T extends ThinEngine>(): T;

```

Comment:
Let s not add this one. The type changes at runtime depending on the running environment.